torch/csrc/utils/tensor_numpy.cpp

```diff
@@ -145,7 +145,7 @@ PyObject* tensor_to_numpy(const at::Tensor& tensor, bool force /*=false*/) {
     TORCH_CHECK(
         !(at::GradMode::is_enabled() && tensor.requires_grad()),
         "Can't call numpy() on Tensor that requires grad. "
-        "Use tensor.detach().numpy() instead.");
+        "Use tensor.detach().cpu().numpy() instead.");
```

Comment:
@ezyang - I think this is getting silly. `tensor.numpy()` should just give you the numpy array. If it needs to make a copy, such so be it.

The lesson I took from `.reshape()` and `.view()` is that APIs that just do the desired operation (`reshape()`) are a much better experience than erroring out if the operation requires a copy. The guaranteed aliasing that `.view()` and `.numpy()` is not important enough to make the general experience worse.

Reply:
Well, the fortunate thing here is, because we are raising an error, it is easy to make it implicitly convert now without breaking BC :)

Reply:
I'm not sure about this change, this error is specifically about the requires_grad part. If you want an all encompassing fix here, i would suggest `tensor.numpy(force=True)` instead of the partial version here